python/pyspark/context.py

```diff
@@ -227,7 +227,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         os.environ["SPARK_BUFFER_SIZE"] = \
             str(self._jvm.PythonUtils.getSparkBufferSize(self._jsc))
 
-        self.pythonExec = os.environ.get("PYSPARK_PYTHON", 'python3')
+        self.pythonExec = os.environ.get("PYSPARK_PYTHON") or self.environment.get("PYSPARK_PYTHON", 'python3')
```

Comment:
I think `environment` argument is meant to be for workder nodes. Probably we should just stick to the semantic 

https://github.com/apache/spark/pull/31997/files#diff-1d250699e56249e8bc181a6c531817f4aed14120fe1daf04ced83dc947dddc0aR81-R83